src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
 export interface Figure {
-
+  shape: string;
+  color: string;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = 'triangle';
+
+  color: string;
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: string, a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || b + c <= a || c + a <= b) {
+      throw new Error('Invalid sides for a triangle');
+    }
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
 
+    return Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+  }
 }
 
-export class Circle {
+export class Circle implements Figure {
+  shape = 'circle';
 
+  color: string;
+
+  radius: number;
+
+  constructor(color: string, radius: number) {
```

Comment:
you can access Figure types with this syntax: `Figure['typeName']` - you won't have to retype everything again